provider/azure/environ.go

```diff
@@ -371,11 +371,12 @@ func (env *azureEnviron) ConstraintsValidator(ctx context.ProviderCallContext) (
 	if err != nil {
 		return nil, err
 	}
-	instTypeNames := make([]string, 0, len(instanceTypes))
-	for instTypeName := range instanceTypes {
-		instTypeNames = append(instTypeNames, instTypeName)
+	instTypes := convertInstanceTypeMapToSlice(instanceTypes)
```

Comment:
@manadart did you mean it like this? So I exchanged the existing azure implementation to use the newly added sorting. 
For this, to work I had to convert the map[string]InstanceType to a slice[]instanceType